File: VB/WindowsApplication3/Data/Record.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Linq
Imports System.Text

Namespace DXSample.Data

    Public Class Record

        Public Sub New()
        End Sub

        Public Property ID As Integer

        Public Property ParentID As Integer

        Public Property Text As String

        Public Property Dt As DateTime

        Public Property State As Boolean

        Public Property Image As Image

        Public Property TestDetails As BindingList(Of DXSample.Data.Detail)

        Public Property Details As BindingList(Of DXSample.Data.Detail)
    End Class

    Public Class Detail

        Public Property ID As Integer

        Public Property Text As String

        Public Property Info As String
    End Class

    Public Class DataHelper

        Public Shared Function GetData(ByVal count As Integer, ByVal detailCount As Integer) As BindingList(Of DXSample.Data.Record)
            Dim records As System.ComponentModel.BindingList(Of DXSample.Data.Record) = New System.ComponentModel.BindingList(Of DXSample.Data.Record)()
            Dim details As System.ComponentModel.BindingList(Of DXSample.Data.Detail) = New System.ComponentModel.BindingList(Of DXSample.Data.Detail)()
            Dim testDetails As System.ComponentModel.BindingList(Of DXSample.Data.Detail) = New System.ComponentModel.BindingList(Of DXSample.Data.Detail)()
            For j As Integer = 0 To detailCount - 1
                details.Add(New DXSample.Data.Detail() With {.ID = j, .Text = String.Format("Text {0}", j), .Info = String.Format("Info {0}", j)})
                testDetails.Add(New DXSample.Data.Detail() With {.ID = j, .Text = String.Format("Test Text {0}", j), .Info = String.Format(" Test Info {0}", j)})
            Next

            For i As Integer = 0 To count - 1
                records.Add(New DXSample.Data.Record() With {.ID = i, .Text = String.Format("Text {0}", i), .Dt = System.DateTime.Now.AddDays(i), .State = i Mod 2 = 0, .Image = System.Drawing.SystemIcons.Information.ToBitmap(), .Details = details, .TestDetails = testDetails})
            Next

            Return records
        End Function

        Public Shared Function GetData(ByVal count As Integer) As BindingList(Of DXSample.Data.Record)
            Dim records As System.ComponentModel.BindingList(Of DXSample.Data.Record) = New System.ComponentModel.BindingList(Of DXSample.Data.Record)()
            For i As Integer = 0 To count - 1
                records.Add(New DXSample.Data.Record() With {.ID = i, .ParentID = i Mod 5, .Text = String.Format("Text {0}", i), .Dt = System.DateTime.Now.AddDays(i), .State = i Mod 2 = 0, .Image = System.Drawing.SystemIcons.Information.ToBitmap()})
            Next

            Return records
        End Function

        Public Shared Function GetDetailData(ByVal count As Integer) As BindingList(Of DXSample.Data.Detail)
            Dim records As System.ComponentModel.BindingList(Of DXSample.Data.Detail) = New System.ComponentModel.BindingList(Of DXSample.Data.Detail)()
            For i As Integer = 0 To count - 1
                records.Add(New DXSample.Data.Detail() With {.ID = i, .Text = String.Format("Text Text Text Text Text Text Text Text Text Text Text Text Text{0}", i), .Info = String.Format("Info {0}", i)})
            Next

            Return records
        End Function
    End Class
End Namespace

File: CS/WindowsApplication3/Data/Record.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Linq;
using System.Text;

namespace DXSample.Data {
    public class Record {
        public Record() {
        }
        public int ID { get; set; }
        public int ParentID { get; set; }
        public string Text { get; set; }
        public DateTime Dt { get; set; }
        public bool State { get; set; }
        public Image Image { get; set; }
        public BindingList<Detail> TestDetails { get; set; }
        public BindingList<Detail> Details { get; set; }
    }

    public class Detail {
        public int ID { get; set; }
        public string Text { get; set; }
        public string Info { get; set; }
    }

    public class DataHelper {
        public static BindingList<Record> GetData(int count, int detailCount){
            BindingList<Record> records = new BindingList<Record>();
            BindingList<Detail> details = new BindingList<Detail>();
            BindingList<Detail> testDetails = new BindingList<Detail>();
            for (int j = 0; j < detailCount; j++) {
                details.Add(new Detail() {
                    ID = j,
                    Text = string.Format("Text {0}", j),
                    Info = string.Format("Info {0}", j)
                });
                testDetails.Add(new Detail() {
                    ID = j,
                    Text = string.Format("Test Text {0}", j),
                    Info = string.Format(" Test Info {0}", j)
                });
            }
            for (int i = 0; i < count; i++)
                records.Add(new Record() { ID = i, 
                    Text = string.Format("Text {0}", i),
                    Dt = DateTime.Now.AddDays(i), 
                    State = i % 2 == 0, 
                    Image = SystemIcons.Information.ToBitmap(), 
                    Details = details,
                    TestDetails = testDetails
                });

            return records;
        }
        public static BindingList<Record> GetData(int count) {
            BindingList<Record> records = new BindingList<Record>();
            for (int i = 0; i < count; i++)
                records.Add(new Record() {
                    ID = i,
                    ParentID = i % 5,
                    Text = string.Format("Text {0}", i),
                    Dt =  DateTime.Now.AddDays(i),
                    State = i % 2 == 0,
                    Image = SystemIcons.Information.ToBitmap(),
                });
            return records;
        }
        public static BindingList<Detail> GetDetailData(int count) {
            BindingList<Detail> records = new BindingList<Detail>();
            for (int i = 0; i < count; i++)
                records.Add(new Detail() {
                    ID = i,
                    Text = string.Format("Text Text Text Text Text Text Text Text Text Text Text Text Text{0}", i),
                    Info = string.Format("Info {0}", i)
                });
            return records;
        }
    }
}
